File: SimpleGallery/app/src/main/java/com/example/simplegalleryapp/ImageAdapter.java
package com.example.simplegalleryapp;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;


public class ImageAdapter extends BaseAdapter {
    private final Context mContext;


    // 1
    public ImageAdapter(Context context,) {
        this.mContext = context;
    }

    // 2
    @Override
    public int getCount() {
        return books.length;
    }

    // 3
    @Override
    public long getItemId(int position) {
        return 0;
    }

    // 4
    @Override
    public Object getItem(int position) {
        return null;
    }

    // 5
    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        TextView dummyTextView = new TextView(mContext);
        dummyTextView.setText(String.valueOf(position));
        return dummyTextView;
    }
}

File: ImagePuzzle/app/src/main/java/com/karol/imagepuzzle/ImageAdapter.kt
package com.karol.imagepuzzle

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.AbsListView
import android.widget.BaseAdapter
import android.widget.ImageView


class ImageAdapter(c:Context, imagesIDArray: ArrayList<Int>, columnsNum: Int) : BaseAdapter() {
    private val context = c
    private val idArray = imagesIDArray
    private val columns = columnsNum
    private val padding = 3

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {

        val imageView: ImageView = ImageView(context)
        imageView.setImageResource(idArray[position])
        if (parent != null) {
                imageView.layoutParams = AbsListView.LayoutParams((0.9*parent.width/columns).toInt(), (0.9*parent.width/columns).toInt())
            }
        imageView.scaleType = ImageView.ScaleType.CENTER_CROP
        imageView.setPadding(padding, padding, padding, padding)
        return imageView
    }

    override fun getItem(position: Int): Any? {
        return null;
    }

    override fun getItemId(position: Int): Long {
        return 0
    }

    override fun getCount(): Int {
        return idArray.lastIndex
    }

}